docs/keystone-auth.md

```diff
@@ -0,0 +1,50 @@
+An authentication backend for Openstack's Keystone is maintained in this
+source tree as `haas.ext.auth.keystone`. This document describes its
+configuration and usage in detail.
+
+NOTE: The HaaS command line interface only supports the keystone v3 API.
+The server supports anything supported by [keystonemiddleware][1].
+
+# Usage
+
+Once HaaS has been configured to work with Keystone, an administrator
+must manually add Openstack projects to HaaS before they can access the
+HaaS API. The HaaS project names must correspond to the Openstack UUIDs.
+For example, an administrator may execute the command:
+
+    haas project_create 00de7c85e594473db7461cdf7367166a
+
+To grant the Openstack project with that UUID access to HaaS.
+
+Note that the plugin recognizes any user with an `admin` role on any
+project as a HaaS administrator, similar to the default policy for core
```

Comment:
That just isn't true.  In both Openstack and haas, 'admin' refers to a global admin (or at least the admin of a domain), and there is no concept of a project admin.